Current State Of Linux Builds

Linux builds of Cemu are not recommended yet for beginners, since currently it lacks a few things:
- There's only prebuilt binaries for ubuntu 20.04, which means that for other distros (or versions) you'll have to [compile it yourself](https://github.com/cemu-project/Cemu/blob/main/BUILD.md).
  - In the future the plan is to also distribute appimages and/or flatpak releases using the CI.
- Cemu currently requires all files to be placed in a case-insensitive filesystem
- Cemu utilizes GTK3 using wxWidgets and xlib for it's Linux port currently. Wayland support should work but other environments aren't tested.

You can use this issue to share your experiences, issues or build tips!

I sadly couldn't get this to compile on Arch so I'm sharing my steps in case somebody else wants to give it a whirl:

1. Install dependencies
```
sudo pacman -S --needed gtk3 libsecret libgcrypt systemd-libs freeglut nasm
```

2. I had `clang-14.0.6-2` installed on my system so I decided to downgrade everything to v12 just in case
```
yay -S downgrade
sudo downgrade clang llvm llvm-libs
```

3. Install VulkanSDK
```
cd ~
mkdir vulkan
cd vulkan
curl https://sdk.lunarg.com/sdk/download/1.3.224.0/linux/vulkansdk-linux-x86_64-1.3.224.0.tar.gz | tar xz
cd 1.3.224.0
source setup-env.sh
```

4. Get Cemu
```
cd ~
git clone --recursive https://github.com/cemu-project/Cemu
cd Cemu
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang-12 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-12 -G Ninja -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja
```

5. Compile
```
ninja
```

6. First issue
```
FAILED: dependencies/cubeb/CMakeFiles/cubeb.dir/src/cubeb_resampler.cpp.o
/usr/bin/clang++ -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XLIB_KHR -D_UNICODE -D__WXGTK3__ -D__WXGTK__ -I/home/user/Cemu/build/src/Cafe -I/home/user/Cemu/src/Cafe -I/home/user/vulkan/1.3.224.0/x86_64/include -I/home/user/Cemu/src/Cafe/. -I/home/user/Cemu/src/Cafe/.. -I/home/user/Cemu/dependencies/cubeb/include -I/home/user/Cemu/build/exports -I/home/user/Cemu/dependencies/ZArchive/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/SDL2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/lib/wx/include/gtk3-unicode-static-3.2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/wx-3.2 -O3 -DNDEBUG   -fms-extensions -fms-compatibility-version=19.14 -fdelayed-template-parsing -maes -Wno-switch -Wno-ignored-attributes -Wno-deprecated-enum-enum-conversion -Wno-ambiguous-reversed-operator -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx -MD -MT src/Cafe/CMakeFiles/CemuCafe.dir/Account/Account.cpp.o -MF src/Cafe/CMakeFiles/CemuCafe.dir/Account/Account.cpp.o.d -o src/Cafe/CMakeFiles/CemuCafe.dir/Account/Account.cpp.o -c /home/user/Cemu/src/Cafe/Account/Account.cpp
In file included from <built-in>:451:
In file included from /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx:5:
In file included from /home/user/Cemu/src/Cafe/../Common/precompiled.h:10:
In file included from /home/user/Cemu/build/vcpkg_installed/x64-linux/include/fmt/core.h:13:
In file included from /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/functional:64:
In file included from /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_algo.h:65:
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/uniform_int_dist.h:250:31: error: expected unqualified-id
          static_assert(!_Up_traits::__is_signed, "U must be unsigned");
                                     ^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/uniform_int_dist.h:251:31: error: expected unqualified-id
          static_assert(!_Wp_traits::__is_signed, "W must be unsigned");
                                     ^
2 errors generated.
```

**Temporary solution**: Turn lines 250 and 251 into comments in `/usr/include/c++/12.2.0/bits/uniform_int_dist.h`

7. Second issue
```
FAILED: src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o 
/usr/bin/clang++ -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XLIB_KHR -D_UNICODE -D__WXGTK3__ -D__WXGTK__ -I/home/user/Cemu/build/src/Cafe -I/home/user/Cemu/src/Cafe -I/home/user/vulkan/1.3.224.0/x86_64/include -I/home/user/Cemu/src/Cafe/. -I/home/user/Cemu/src/Cafe/.. -I/home/user/Cemu/dependencies/cubeb/include -I/home/user/Cemu/build/exports -I/home/user/Cemu/dependencies/ZArchive/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/SDL2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/lib/wx/include/gtk3-unicode-static-3.2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/wx-3.2 -O3 -DNDEBUG   -fms-extensions -fms-compatibility-version=19.14 -fdelayed-template-parsing -maes -Wno-switch -Wno-ignored-attributes -Wno-deprecated-enum-enum-conversion -Wno-ambiguous-reversed-operator -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx -MD -MT src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -MF src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o.d -o src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -c /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp
/home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
#include <SPIRV/Logger.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
```

**Workaround:** Change line 10 into `#if true` in `~/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp`

8. Third issue
```
FAILED: src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o 
/usr/bin/clang++ -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XLIB_KHR -D_UNICODE -D__WXGTK3__ -D__WXGTK__ -I/home/user/Cemu/build/src/Cafe -I/home/user/Cemu/src/Cafe -I/home/user/vulkan/1.3.224.0/x86_64/include -I/home/user/Cemu/src/Cafe/. -I/home/user/Cemu/src/Cafe/.. -I/home/user/Cemu/dependencies/cubeb/include -I/home/user/Cemu/build/exports -I/home/user/Cemu/dependencies/ZArchive/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/SDL2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/lib/wx/include/gtk3-unicode-static-3.2 -isystem /home/user/Cemu/build/vcpkg_installed/x64-linux/include/wx-3.2 -O3 -DNDEBUG   -fms-extensions -fms-compatibility-version=19.14 -fdelayed-template-parsing -maes -Wno-switch -Wno-ignored-attributes -Wno-deprecated-enum-enum-conversion -Wno-ambiguous-reversed-operator -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /home/user/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx -MD -MT src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -MF src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o.d -o src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -c /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp
In file included from /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:19:
In file included from /home/user/Cemu/src/Cafe/../Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.h:5:
In file included from /home/user/Cemu/src/Cafe/../Cafe/HW/Latte/Renderer/Vulkan/LatteTextureVk.h:7:
/home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:134:18: error: excess elements in scalar initializer
        /* .limits = */ {
                        ^
1 error generated.
```

**Workaround:** Remove the curly brackets in line 134 and 144 in `~/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp`

9. Final boss
```
FAILED: /home/user/Cemu/bin/Cemu 
/usr/bin/clang++ -O3 -DNDEBUG  src/CMakeFiles/CemuBin.dir/main.cpp.o src/CMakeFiles/CemuBin.dir/mainLLE.cpp.o -o /home/user/Cemu/bin/Cemu  vcpkg_installed/x64-linux/lib/libboost_program_options.a  vcpkg_installed/x64-linux/lib/libboost_filesystem.a  vcpkg_installed/x64-linux/lib/libboost_nowide.a  /usr/lib/libSM.so  /usr/lib/libICE.so  /usr/lib/libX11.so  /usr/lib/libXext.so  vcpkg_installed/x64-linux/lib/libSDL2.a  vcpkg_installed/x64-linux/lib/libSDL2main.a  vcpkg_installed/x64-linux/lib/libSDL2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  dependencies/cubeb/libcubeb.a  src/Common/libCemuCommon.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/config/libCemuConfig.a  src/gui/libCemuGui.a  src/audio/libCemuAudio.a  src/input/libCemuInput.a  src/util/libCemuUtil.a  src/asm/libCemuAsm.a  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libz.a  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_adv-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_aui-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_html-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_propgrid-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_ribbon-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_stc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_xrc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_gl-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_qa-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu_xml-3.2.a  vcpkg_installed/x64-linux/lib/libcurl.a  vcpkg_installed/x64-linux/lib/libimgui.a  vcpkg_installed/x64-linux/lib/libpugixml.a  src/Common/libCemuCommon.a  src/audio/libCemuAudio.a  src/input/libCemuInput.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/config/libCemuConfig.a  src/gui/libCemuGui.a  src/imgui/libimguiImpl.a  src/Common/libCemuCommon.a  src/input/libCemuInput.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/resource/libCemuResource.a  src/gui/libCemuGui.a  src/asm/libCemuAsm.a  src/Cemu/libCemuComponents.a  dependencies/discord-rpc/src/libdiscord-rpc.a  -lpthread  src/Cafe/libCemuCafe.a  vcpkg_installed/x64-linux/lib/libglslang.a  vcpkg_installed/x64-linux/lib/libSPIRV.a  vcpkg_installed/x64-linux/lib/libMachineIndependent.a  vcpkg_installed/x64-linux/lib/libOGLCompiler.a  vcpkg_installed/x64-linux/lib/libOSDependent.a  vcpkg_installed/x64-linux/lib/libGenericCodeGen.a  dependencies/ih264d/libih264d.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_adv-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_aui-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_propgrid-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_ribbon-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_stc-3.2.a  vcpkg_installed/x64-linux/lib/libwxscintilla-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_xrc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_html-3.2.a  /usr/lib/libmspack.so  vcpkg_installed/x64-linux/lib/libwx_gtk3u_gl-3.2.a  /usr/lib/libGLU.so  /usr/lib/libGLX.so  /usr/lib/libEGL.so  /usr/lib/libOpenGL.so  -lffi  vcpkg_installed/x64-linux/lib/libwx_gtk3u_qa-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu_xml-3.2.a  src/resource/libCemuResource.a  dependencies/ZArchive/libzarchive.a  vcpkg_installed/x64-linux/lib/libzstd.a  vcpkg_installed/x64-linux/lib/libzip.a  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libz.a  vcpkg_installed/x64-linux/lib/libcrypto.a  vcpkg_installed/x64-linux/lib/libpugixml.a  dependencies/cubeb/libcubeb.a  vcpkg_installed/x64-linux/lib/libboost_program_options.a  vcpkg_installed/x64-linux/lib/libboost_filesystem.a  vcpkg_installed/x64-linux/lib/libboost_nowide.a  vcpkg_installed/x64-linux/lib/libSDL2.a  -lwayland-egl  -lwayland-client  -lwayland-cursor  -lEGL  -lxkbcommon  /usr/lib/libXss.so  /usr/lib/libXxf86vm.so  -ldbus-1  -lunwind  -lunwind-generic  -lrt  -Wl,--no-undefined  -pthread  vcpkg_installed/x64-linux/lib/libSDL2main.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  -lm  /usr/lib/libgtk-3.so  /usr/lib/libgdk-3.so  /usr/lib/libz.so  -lm  /usr/lib/libpcre.so  /usr/lib/libffi.so  /usr/lib/libbz2.so  /usr/lib/libbrotlicommon.so  /usr/lib/libgraphite2.so  /usr/lib/libglib-2.0.so  /usr/lib/libgio-2.0.so  /usr/lib/libgobject-2.0.so  /usr/lib/libgmodule-2.0.so  /usr/lib/libblkid.so  /usr/lib/libfribidi.so  /usr/lib/libdatrie.so  /usr/lib/libfreetype.so  /usr/lib/libbrotlidec.so  /usr/lib/libXrender.so  /usr/lib/libpthread.a  /usr/lib/libxcb.so  /usr/lib/libXau.so  /usr/lib/libdl.a  /usr/lib/liblzo2.so  /usr/lib/libpangocairo-1.0.so  /usr/lib/libpango-1.0.so  /usr/lib/libcairo.so  /usr/lib/libharfbuzz-gobject.so  /usr/lib/libfontconfig.so  vcpkg_installed/x64-linux/lib/libpng16.a  /usr/lib/libcairo-gobject.so  /usr/lib/libXcomposite.so  /usr/lib/libXdamage.so  /usr/lib/libwayland-client.so  /usr/lib/librt.a  /usr/lib/libepoxy.so  /usr/lib/libcloudproviders.so  /usr/lib/libatk-1.0.so  /usr/lib/libgdk_pixbuf-2.0.so  vcpkg_installed/x64-linux/lib/libtiff.a  vcpkg_installed/x64-linux/lib/libjpeg.a  vcpkg_installed/x64-linux/lib/liblzma.a  /usr/lib/libatk-bridge-2.0.so  /usr/lib/libsystemd.so  /usr/lib/libatspi.so  /usr/lib/libdbus-1.so  /usr/lib/libXtst.so  /usr/lib/libxkbcommon.so  /usr/lib/libwayland-cursor.so  /usr/lib/libwayland-egl.so  /usr/lib/libGL.so  /usr/lib/libEGL.so  /usr/lib/libpangoft2-1.0.so  /usr/lib/libthai.so  /usr/lib/libXft.so  /usr/lib/libxcb-render.so  /usr/lib/libxcb-shm.so  /usr/lib/libXdmcp.so  /usr/lib/libpixman-1.so  /usr/lib/libharfbuzz.so  /usr/lib/libmount.so  /usr/lib/libsysprof-capture-4.a  /usr/lib/libz.so  -lm  /usr/lib/libpcre.so  /usr/lib/libffi.so  /usr/lib/libbz2.so  /usr/lib/libbrotlicommon.so  /usr/lib/libgraphite2.so  /usr/lib/libglib-2.0.so  /usr/lib/libgio-2.0.so  /usr/lib/libgobject-2.0.so  /usr/lib/libgmodule-2.0.so  /usr/lib/libblkid.so  /usr/lib/libfribidi.so  /usr/lib/libdatrie.so  /usr/lib/libfreetype.so  /usr/lib/libbrotlidec.so  /usr/lib/libXrender.so  /usr/lib/libpthread.a  /usr/lib/libxcb.so  /usr/lib/libXau.so  /usr/lib/libdl.a  /usr/lib/liblzo2.so  /usr/lib/libpangocairo-1.0.so  /usr/lib/libpango-1.0.so  /usr/lib/libcairo.so  /usr/lib/libharfbuzz-gobject.so  /usr/lib/libfontconfig.so  vcpkg_installed/x64-linux/lib/libpng16.a  /usr/lib/libcairo-gobject.so  /usr/lib/libXcomposite.so  /usr/lib/libXdamage.so  /usr/lib/libwayland-client.so  /usr/lib/librt.a  /usr/lib/libepoxy.so  /usr/lib/libcloudproviders.so  /usr/lib/libatk-1.0.so  /usr/lib/libgdk_pixbuf-2.0.so  vcpkg_installed/x64-linux/lib/libtiff.a  vcpkg_installed/x64-linux/lib/libjpeg.a  vcpkg_installed/x64-linux/lib/liblzma.a  /usr/lib/libatk-bridge-2.0.so  /usr/lib/libsystemd.so  /usr/lib/libatspi.so  /usr/lib/libdbus-1.so  /usr/lib/libXtst.so  /usr/lib/libxkbcommon.so  /usr/lib/libwayland-cursor.so  /usr/lib/libwayland-egl.so  /usr/lib/libGL.so  /usr/lib/libEGL.so  /usr/lib/libpangoft2-1.0.so  /usr/lib/libthai.so  /usr/lib/libXft.so  /usr/lib/libxcb-render.so  /usr/lib/libxcb-shm.so  /usr/lib/libXdmcp.so  /usr/lib/libpixman-1.so  /usr/lib/libharfbuzz.so  /usr/lib/libmount.so  /usr/lib/libsysprof-capture-4.a  /usr/lib/libXcursor.so  /usr/lib/libXrandr.so  /usr/lib/libXinerama.so  /usr/lib/libXi.so  /usr/lib/libXfixes.so  vcpkg_installed/x64-linux/lib/libexpat.a  /usr/lib/libSM.so  /usr/lib/libICE.so  /usr/lib/libX11.so  /usr/lib/libXext.so  /usr/lib/libgspell-1.so  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  vcpkg_installed/x64-linux/lib/libz.a  -lc  -ldl  /usr/lib/libspeexdsp.so
```
```
/usr/bin/ld: src/CMakeFiles/CemuBin.dir/main.cpp.o: in function `std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> > fmt::v7::detail::vformat<char8_t, 0>(fmt::v7::basic_string_view<char8_t>, fmt::v7::basic_format_args<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<fmt::v7::type_identity<char8_t>::type>, fmt::v7::type_identity<char8_t>::type> >)':
main.cpp:(.text._ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE[_ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE]+0x6e): undefined reference to `void std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::_M_construct<char8_t const*>(char8_t const*, char8_t const*, std::forward_iterator_tag)'
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

Fedora is stuck here

`CMake Warning at buildtrees/versioning_/versions/openssl/557ff31f9a64f01cd0d98dd44793ce3c7fd32893/portfile.cmake:7 (message):
  openssl currently requires the following library from the system package
  manager:

      linux-headers

  It can be installed on alpine systems via apk add linux-headers.
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)


-- Using cached openssl-openssl-openssl-3.0.5.tar.gz.
-- Cleaning sources at /home/matt/Cemu/dependencies/vcpkg/buildtrees/openssl/src/nssl-3.0.5-30ecfd0d02.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/matt/Cemu/dependencies/vcpkg/downloads/openssl-openssl-openssl-3.0.5.tar.gz
-- Using source at /home/matt/Cemu/dependencies/vcpkg/buildtrees/openssl/src/nssl-3.0.5-30ecfd0d02.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:129 (message):
    Command failed: /home/matt/Cemu/dependencies/vcpkg/downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j13
    Working Directory: /home/matt/Cemu/dependencies/vcpkg/buildtrees/openssl/x64-linux-dbg
    See logs for more information:
      /home/matt/Cemu/dependencies/vcpkg/buildtrees/openssl/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  /home/matt/Cemu/build/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /home/matt/Cemu/build/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  buildtrees/versioning_/versions/openssl/557ff31f9a64f01cd0d98dd44793ce3c7fd32893/unix/portfile.cmake:21 (vcpkg_cmake_install)
  buildtrees/versioning_/versions/openssl/557ff31f9a64f01cd0d98dd44793ce3c7fd32893/portfile.cmake:39 (include)
  scripts/ports.cmake:147 (include)


error: building openssl:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openssl
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[openssl]+Build+error
Include '[openssl] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-07-21-a0e87e227afb536c62188c11ad029954f28fdb22
    vcpkg-scripts version: 1b0252ca 2022-08-22 (2 days ago)

You can also use the prefilled template from /home/matt/Cemu/build/vcpkg_installed/vcpkg/issue_body.md.

-- Running vcpkg install - failed
CMake Error at dependencies/vcpkg/scripts/buildsystems/vcpkg.cmake:860 (message):
  vcpkg install failed.  See logs for more information:
  /home/matt/Cemu/build/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/share/cmake/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:18 (project)
`

So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
```
/home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
#include "imgui.h"
         ^~~~~~~~~
/home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
#include "imgui.h"
         ^~~~~~~~~
In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
In file included from /usr/include/wx-3.2/wx/math.h:16:
In file included from /usr/include/wx-3.2/wx/defs.h:45:
/usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
#include "wx/setup.h"
         ^~~~~~~~~~~~
```
So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.

So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?

Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
```
	-DCMAKE_CXX_STANDARD=20 
	-DCMAKE_C_STANDARD=17 
```

> So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
> 
> ```
> /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
> #include "imgui.h"
>          ^~~~~~~~~
> /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
> #include "imgui.h"
>          ^~~~~~~~~
> In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
> In file included from /usr/include/wx-3.2/wx/math.h:16:
> In file included from /usr/include/wx-3.2/wx/defs.h:45:
> /usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
> #include "wx/setup.h"
>          ^~~~~~~~~~~~
> ```
> 
> So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.
> 
> So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?
> 
> Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
> 
> ```
> 	-DCMAKE_CXX_STANDARD=20 
> 	-DCMAKE_C_STANDARD=17 
> ```
> 
> Here is my cmake command btw
> 
> ```
> 	cmake \
> 	../src \
> 	-DCMAKE_BUILD_TYPE=Release \
> 	-DCMAKE_MAKE_PROGRAM=/usr/bin/make \
> 	-DCMAKE_C_COMPILER=clang \
> 	-DCMAKE_CXX_COMPILER=clang++ \
> 	-DCMAKE_CXX_STANDARD=20 \
> 	-DCMAKE_C_STANDARD=17 \
> 	-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx" \
> 	-DCMAKE_C_FLAGS="$CFLAGS -I/usr/include/wx"
> ```
> 
> And, here are the build dependencies and make dependencies
> 
> ```
> depends=('vulkan-driver' 'opengl-driver' 'gtk3' 'libsecret' 'libgcrypt' 'freeglut' 'glibc' 'openssl' 'zlib' 'curl' 'pugixml' 'glslang' 'x264' 'libzip')
> makedepends=('cmake' 'vulkan-headers' 'spirv-tools' 'vulkan-validation-layers' 'nasm' 'boost' 'glm' 'wxwidgets-common')
> ```
I am not a member of the cemu team, however I imagine reporting issues with other peoples packages  is not helpful. I would comment on the aur package

> So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
> 
> ```
> /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
> #include "imgui.h"
>          ^~~~~~~~~
> /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
> #include "imgui.h"
>          ^~~~~~~~~
> In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
> In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
> In file included from /usr/include/wx-3.2/wx/math.h:16:
> In file included from /usr/include/wx-3.2/wx/defs.h:45:
> /usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
> #include "wx/setup.h"
>          ^~~~~~~~~~~~
> ```
> 
> So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.
> 
> So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?
> 
> Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
> 
> ```
> 	-DCMAKE_CXX_STANDARD=20 
> 	-DCMAKE_C_STANDARD=17 
> ```
> 
> Here is my cmake command btw
> 
> ```
> 	cmake \
> 	../src \
> 	-DCMAKE_BUILD_TYPE=Release \
> 	-DCMAKE_MAKE_PROGRAM=/usr/bin/make \
> 	-DCMAKE_C_COMPILER=clang \
> 	-DCMAKE_CXX_COMPILER=clang++ \
> 	-DCMAKE_CXX_STANDARD=20 \
> 	-DCMAKE_C_STANDARD=17 \
> 	-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx" \
> 	-DCMAKE_C_FLAGS="$CFLAGS -I/usr/include/wx"
> ```
> 
> And, here are the build dependencies and make dependencies
> 
> ```
> depends=('vulkan-driver' 'opengl-driver' 'gtk3' 'libsecret' 'libgcrypt' 'freeglut' 'glibc' 'openssl' 'zlib' 'curl' 'pugixml' 'glslang' 'x264' 'libzip')
> makedepends=('cmake' 'vulkan-headers' 'spirv-tools' 'vulkan-validation-layers' 'nasm' 'boost' 'glm' 'wxwidgets-common')
> ```

Any reason you are using `make` instead of `ninja`?

> > So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
> > ```
> > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
> > #include "imgui.h"
> >          ^~~~~~~~~
> > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
> > #include "imgui.h"
> >          ^~~~~~~~~
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
> > In file included from /usr/include/wx-3.2/wx/math.h:16:
> > In file included from /usr/include/wx-3.2/wx/defs.h:45:
> > /usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
> > #include "wx/setup.h"
> >          ^~~~~~~~~~~~
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.
> > So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?
> > Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
> > ```
> > 	-DCMAKE_CXX_STANDARD=20 
> > 	-DCMAKE_C_STANDARD=17 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Here is my cmake command btw
> > ```
> > 	cmake \
> > 	../src \
> > 	-DCMAKE_BUILD_TYPE=Release \
> > 	-DCMAKE_MAKE_PROGRAM=/usr/bin/make \
> > 	-DCMAKE_C_COMPILER=clang \
> > 	-DCMAKE_CXX_COMPILER=clang++ \
> > 	-DCMAKE_CXX_STANDARD=20 \
> > 	-DCMAKE_C_STANDARD=17 \
> > 	-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx" \
> > 	-DCMAKE_C_FLAGS="$CFLAGS -I/usr/include/wx"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And, here are the build dependencies and make dependencies
> > ```
> > depends=('vulkan-driver' 'opengl-driver' 'gtk3' 'libsecret' 'libgcrypt' 'freeglut' 'glibc' 'openssl' 'zlib' 'curl' 'pugixml' 'glslang' 'x264' 'libzip')
> > makedepends=('cmake' 'vulkan-headers' 'spirv-tools' 'vulkan-validation-layers' 'nasm' 'boost' 'glm' 'wxwidgets-common')
> > ```
> 
> I am not a member of the cemu team, however I imagine reporting issues with other peoples packages is not helpful. I would comment on the aur package

I am writing the PKGBUILD, and I need help. The original `cemu` aur package just packages the .exe and launches it with wine.

> > So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
> > ```
> > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
> > #include "imgui.h"
> >          ^~~~~~~~~
> > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
> > #include "imgui.h"
> >          ^~~~~~~~~
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
> > In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
> > In file included from /usr/include/wx-3.2/wx/math.h:16:
> > In file included from /usr/include/wx-3.2/wx/defs.h:45:
> > /usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
> > #include "wx/setup.h"
> >          ^~~~~~~~~~~~
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.
> > So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?
> > Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
> > ```
> > 	-DCMAKE_CXX_STANDARD=20 
> > 	-DCMAKE_C_STANDARD=17 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Here is my cmake command btw
> > ```
> > 	cmake \
> > 	../src \
> > 	-DCMAKE_BUILD_TYPE=Release \
> > 	-DCMAKE_MAKE_PROGRAM=/usr/bin/make \
> > 	-DCMAKE_C_COMPILER=clang \
> > 	-DCMAKE_CXX_COMPILER=clang++ \
> > 	-DCMAKE_CXX_STANDARD=20 \
> > 	-DCMAKE_C_STANDARD=17 \
> > 	-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx" \
> > 	-DCMAKE_C_FLAGS="$CFLAGS -I/usr/include/wx"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And, here are the build dependencies and make dependencies
> > ```
> > depends=('vulkan-driver' 'opengl-driver' 'gtk3' 'libsecret' 'libgcrypt' 'freeglut' 'glibc' 'openssl' 'zlib' 'curl' 'pugixml' 'glslang' 'x264' 'libzip')
> > makedepends=('cmake' 'vulkan-headers' 'spirv-tools' 'vulkan-validation-layers' 'nasm' 'boost' 'glm' 'wxwidgets-common')
> > ```
> 
> Any reason you are using `make` instead of `ninja`?

Honestly, I just use make to have one less dependency (as it shouldn't change anything with CMake), but I will try with ninja to see if it changes everything

> > > So, I am writing a PKGBUILD for the Arch Linux AUR, and I am currently getting errors linking to wxwidgets and imgui.
> > > ```
> > > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_vulkan.cpp:45:10: fatal error: 'imgui.h' file not found
> > > #include "imgui.h"
> > >          ^~~~~~~~~
> > > /home/hurricane/cemu/src/Cemu/src/imgui/imgui_impl_opengl3.cpp:68:10: fatal error: 'imgui.h' file not found
> > > #include "imgui.h"
> > >          ^~~~~~~~~
> > > In file included from /home/hurricane/cemu/src/Cemu/src/input/emulated/EmulatedController.cpp:1:
> > > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/emulated/EmulatedController.h:6:
> > > In file included from /home/hurricane/cemu/src/Cemu/src/input/../input/motion/MotionSample.h:3:
> > > In file included from /home/hurricane/cemu/src/Cemu/src/input/../util/math/quaternion.h:4:
> > > In file included from /usr/include/wx-3.2/wx/math.h:16:
> > > In file included from /usr/include/wx-3.2/wx/defs.h:45:
> > > /usr/include/wx-3.2/wx/platform.h:159:10: fatal error: 'wx/setup.h' file not found
> > > #include "wx/setup.h"
> > >          ^~~~~~~~~~~~
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > So I looked and say Arch Linux installs its headers for the package `wxwidgets-common` at `/usr/include/wx-3.2/`, so I added to my CMake command `-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx-3.2"`, but then that leads to another error and so forth.
> > > So, is this an issue with how Arch packages its headers for `wxwidgets`, or what?
> > > Also, it looks y'all didn't set your C++ and C standard cause I had to manually set them as I had problems with some things being missing from the `std`.
> > > ```
> > > 	-DCMAKE_CXX_STANDARD=20 
> > > 	-DCMAKE_C_STANDARD=17 
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Here is my cmake command btw
> > > ```
> > > 	cmake \
> > > 	../src \
> > > 	-DCMAKE_BUILD_TYPE=Release \
> > > 	-DCMAKE_MAKE_PROGRAM=/usr/bin/make \
> > > 	-DCMAKE_C_COMPILER=clang \
> > > 	-DCMAKE_CXX_COMPILER=clang++ \
> > > 	-DCMAKE_CXX_STANDARD=20 \
> > > 	-DCMAKE_C_STANDARD=17 \
> > > 	-DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/wx" \
> > > 	-DCMAKE_C_FLAGS="$CFLAGS -I/usr/include/wx"
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And, here are the build dependencies and make dependencies
> > > ```
> > > depends=('vulkan-driver' 'opengl-driver' 'gtk3' 'libsecret' 'libgcrypt' 'freeglut' 'glibc' 'openssl' 'zlib' 'curl' 'pugixml' 'glslang' 'x264' 'libzip')
> > > makedepends=('cmake' 'vulkan-headers' 'spirv-tools' 'vulkan-validation-layers' 'nasm' 'boost' 'glm' 'wxwidgets-common')
> > > ```
> > 
> > 
> > Any reason you are using `make` instead of `ninja`?
> 
> Honestly, I just use make to have one less dependency (as it shouldn't change anything with CMake), but I will try with ninja to see if it changes everything

Trying Ninja stills gives it the same problems
```
[21/390] Building CXX object Cafe/CMakeFiles/CemuCafe.dir/Account/Account.o
FAILED: Cafe/CMakeFiles/CemuCafe.dir/Account/Account.o 
/usr/bin/clang++ -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XLIB_KHR -I/home/hurricane/cemu/src/Cemu/build/Cafe -I/home/hurricane/cemu/src/Cemu/src/Cafe -I/home/hurricane/cemu/src/Cemu/src/Cafe/. -I/home/hurricane/cemu/src/Cemu/src/Cafe/.. -march=znver3 -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -I/usr/include/wx -O3 -DNDEBUG   -fms-extensions -fms-compatibility-version=19.14 -fdelayed-template-parsing -maes -Wno-switch -Wno-ignored-attributes -Wno-deprecated-enum-enum-conversion -Wno-ambiguous-reversed-operator -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /home/hurricane/cemu/src/Cemu/build/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /home/hurricane/cemu/src/Cemu/build/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx -MD -MT Cafe/CMakeFiles/CemuCafe.dir/Account/Account.o -MF Cafe/CMakeFiles/CemuCafe.dir/Account/Account.o.d -o Cafe/CMakeFiles/CemuCafe.dir/Account/Account.o -c /home/hurricane/cemu/src/Cemu/src/Cafe/Account/Account.cpp
In file included from /home/hurricane/cemu/src/Cemu/src/Cafe/Account/Account.cpp:3:
/home/hurricane/cemu/src/Cemu/src/Cafe/../gui/CemuApp.h:3:10: fatal error: 'wx/app.h' file not found
#include <wx/app.h>
         ^~~~~~~~~~
1 error generated.
```

Multiple missing components to compile on Debian, but they can be acquired easily.
clang-12 isn't available on Debian 11. Must be acquired from the llvm repository.
cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source.
Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.

Despite all, Cemu won't compile.
```
/home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
#include <SPIRV/Logger.h>
```

> Multiple missing components to compile on Debian, but they can be acquired easily. clang-12 isn't available on Debian 11. Must be acquired from the llvm repository. cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source. Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.
> 
> Despite all, Cemu won't compile.
> 
> ```
> /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
> #include <SPIRV/Logger.h>
> ```

You need glslang

> > Multiple missing components to compile on Debian, but they can be acquired easily. clang-12 isn't available on Debian 11. Must be acquired from the llvm repository. cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source. Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.
> > Despite all, Cemu won't compile.
> > ```
> > /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
> > #include <SPIRV/Logger.h>
> > ```
> 
> You need glslang for <SPIRV/Logger.h>

glslang-dev and glslang-tools are both already installed

> > > Multiple missing components to compile on Debian, but they can be acquired easily. clang-12 isn't available on Debian 11. Must be acquired from the llvm repository. cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source. Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.
> > > Despite all, Cemu won't compile.
> > > ```
> > > /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
> > > #include <SPIRV/Logger.h>
> > > ```
> > 
> > 
> > You need glslang for <SPIRV/Logger.h>
> 
> glslang-dev and glslang-tools are both already installed (11.1.0)

Check your `/usr/include/` and look to see if you have `/usr/include/glslang/SPIRV/Logger.h`

> > > > Multiple missing components to compile on Debian, but they can be acquired easily. clang-12 isn't available on Debian 11. Must be acquired from the llvm repository. cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source. Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.
> > > > Despite all, Cemu won't compile.
> > > > ```
> > > > /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
> > > > #include <SPIRV/Logger.h>
> > > > ```
> > > 
> > > 
> > > You need glslang for <SPIRV/Logger.h>
> > 
> > 
> > glslang-dev and glslang-tools are both already installed (11.1.0)
> 
> Check your `/usr/include/` and look to see if you have `/usr/include/glslang/SPIRV/Logger.h`

Yes, the `/usr/include/glslang/SPIRV/Logger.h` file is present there.
```
user@debian:~$ dpkg-query -S /usr/include/glslang/SPIRV/Logger.h
glslang-dev:amd64: /usr/include/glslang/SPIRV/Logger.h
```

> > > > > Multiple missing components to compile on Debian, but they can be acquired easily. clang-12 isn't available on Debian 11. Must be acquired from the llvm repository. cmake 3.21 isn't available on Debian 11. Must be compiled and installed from source. Vulkan SDK isn't avilable on Debian 11. Must be compiled and installed from source.
> > > > > Despite all, Cemu won't compile.
> > > > > ```
> > > > > /home/user/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/Logger.h' file not found
> > > > > #include <SPIRV/Logger.h>
> > > > > ```
> > > > 
> > > > 
> > > > You need glslang for <SPIRV/Logger.h>
> > > 
> > > 
> > > glslang-dev and glslang-tools are both already installed (11.1.0)
> > 
> > 
> > Check your `/usr/include/` and look to see if you have `/usr/include/glslang/SPIRV/Logger.h`
> 
> Yes, the `/usr/include/glslang/SPIRV/Logger.h` file is present there.
> 
> ```
> user@debian:~$ dpkg-query -S /usr/include/glslang/SPIRV/Logger.h
> glslang-dev:amd64: /usr/include/glslang/SPIRV/Logger.h
> ```

Then I don't know how to help you on Debian. Maybe the pkg-config file on Debian is modified? You could try adding the CXX/CFLAGS `-I/usr/include/glslang/`

On Arch Linux, with both ninja and make I arrive at linkage stage and then it fails. Example with make:

```
/usr/bin/ld: CMakeFiles/CemuBin.dir/main.cpp.o: in function `std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> > fmt::v7::detail::vformat<char8_t, 0>(fmt::v7::basic_string_view<char8_t>, fmt::v7::basic_format_args<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<fmt::v7::type_identity<char8_t>::type>, fmt::v7::type_identity<char8_t>::type> >)':
main.cpp:(.text._ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE[_ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE]+0x6f): undefined reference to `void std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::_M_construct<char8_t const*>(char8_t const*, char8_t const*, std::forward_iterator_tag)'
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/CMakeFiles/CemuBin.dir/build.make:366: /home/marco/Downloads/Cemu/bin/Cemu] Error 1
make[1]: *** [CMakeFiles/Makefile2:582: src/CMakeFiles/CemuBin.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

i am way past the cmake part only ninja is failing 
it's just not seeing spriv]
`FAILED: src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o 
/usr/bin/clang++ -DENABLE_DISCORD_RPC -DHAS_CUBEB=1 -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_XCB_KHR -DVK_USE_PLATFORM_XLIB_KHR -D_UNICODE -D__WXGTK3__ -D__WXGTK__ -I/home/matt/Cemu/build/src/Cafe -I/home/matt/Cemu/src/Cafe -I/home/matt/vulkanSdk/1.3.224.0/x86_64/include -I/home/matt/Cemu/src/Cafe/. -I/home/matt/Cemu/src/Cafe/.. -I/home/matt/Cemu/dependencies/cubeb/include -I/home/matt/Cemu/build/exports -I/home/matt/Cemu/dependencies/ZArchive/include -isystem /home/matt/Cemu/build/vcpkg_installed/x64-linux/include -isystem /home/matt/Cemu/build/vcpkg_installed/x64-linux/include/SDL2 -isystem /home/matt/Cemu/build/vcpkg_installed/x64-linux/lib/wx/include/gtk3-unicode-static-3.2 -isystem /home/matt/Cemu/build/vcpkg_installed/x64-linux/include/wx-3.2 -O3 -DNDEBUG   -fms-extensions -fms-compatibility-version=19.14 -fdelayed-template-parsing -maes -Wno-switch -Wno-ignored-attributes -Wno-deprecated-enum-enum-conversion -Wno-ambiguous-reversed-operator -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /home/matt/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /home/matt/Cemu/build/src/Cafe/CMakeFiles/CemuCafe.dir/cmake_pch.hxx -MD -MT src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -MF src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o.d -o src/Cafe/CMakeFiles/CemuCafe.dir/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp.o -c /home/matt/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp
/home/matt/Cemu/src/Cafe/HW/Latte/Renderer/Vulkan/RendererShaderVk.cpp:13:10: fatal error: 'SPIRV/GlslangToSpv.h' file not found
#include <SPIRV/GlslangToSpv.h>
`

I'm trying to compile this on Fedora 36 with ```clang-14```
So I followed this guide (I think issue one was the only issue that affected the build)
https://github.com/cemu-project/Cemu/issues/1#issuecomment-1225289477

And I ran into the same linker issue
```
[482/482] Linking CXX executable ../bin/Cemu
FAILED: ../bin/Cemu 
: && /usr/bin/clang++ -O3 -DNDEBUG  src/CMakeFiles/CemuBin.dir/main.cpp.o src/CMakeFiles/CemuBin.dir/mainLLE.cpp.o -o 
...
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed
```
Nice....we're almost there!

I must have missed something. How were people able to resolve the error

> In file included from /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/stl_algo.h:65:
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/uniform_int_dist.h:250:31: error: expected unqualified-id
          static_assert(!_Up_traits::__is_signed, "U must be unsigned");
                                     ^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/uniform_int_dist.h:251:31: error: expected unqualified-id
          static_assert(!_Wp_traits::__is_signed, "W must be unsigned");
                                     ^
2 errors generated.

I'm getting a final linker error on Pop_OS 22.04. 

```
[481/481] Linking CXX executable src/Cemu
FAILED: src/Cemu 
...
/usr/bin/ld: cannot open output file src/Cemu: Is a directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.

```
Looks like it's wanting to put the binary into a file that already exists, but it's a directory instead so it just fails. Any fix for this?

I get all the way to end of compilation then get these errors:

[485/485] Linking CXX executable ../bin/Cemu
FAILED: ../bin/Cemu 
: && /usr/bin/clang++-12 -O3 -DNDEBUG  src/CMakeFiles/CemuBin.dir/main.cpp.o src/CMakeFiles/CemuBin.dir/mainLLE.cpp.o -o ../bin/Cemu  -Wl,-rpath,/usr/local/lib/x86_64-linux-gnu:/usr/local/lib  vcpkg_installed/x64-linux/lib/libboost_program_options.a  vcpkg_installed/x64-linux/lib/libboost_filesystem.a  vcpkg_installed/x64-linux/lib/libboost_nowide.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  vcpkg_installed/x64-linux/lib/libSDL2.a  vcpkg_installed/x64-linux/lib/libSDL2main.a  vcpkg_installed/x64-linux/lib/libSDL2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  dependencies/cubeb/libcubeb.a  src/Common/libCemuCommon.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/config/libCemuConfig.a  src/gui/libCemuGui.a  src/audio/libCemuAudio.a  src/input/libCemuInput.a  src/util/libCemuUtil.a  src/asm/libCemuAsm.a  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libz.a  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_adv-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_aui-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_html-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_propgrid-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_ribbon-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_stc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_xrc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_gl-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_qa-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu_xml-3.2.a  vcpkg_installed/x64-linux/lib/libcurl.a  vcpkg_installed/x64-linux/lib/libimgui.a  vcpkg_installed/x64-linux/lib/libpugixml.a  src/Common/libCemuCommon.a  src/audio/libCemuAudio.a  src/input/libCemuInput.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/config/libCemuConfig.a  src/gui/libCemuGui.a  src/imgui/libimguiImpl.a  src/Common/libCemuCommon.a  src/input/libCemuInput.a  src/Cemu/libCemuComponents.a  src/Cafe/libCemuCafe.a  src/resource/libCemuResource.a  src/gui/libCemuGui.a  src/asm/libCemuAsm.a  src/Cemu/libCemuComponents.a  dependencies/discord-rpc/src/libdiscord-rpc.a  -lpthread  src/Cafe/libCemuCafe.a  vcpkg_installed/x64-linux/lib/libglslang.a  vcpkg_installed/x64-linux/lib/libSPIRV.a  vcpkg_installed/x64-linux/lib/libMachineIndependent.a  vcpkg_installed/x64-linux/lib/libOGLCompiler.a  vcpkg_installed/x64-linux/lib/libOSDependent.a  vcpkg_installed/x64-linux/lib/libGenericCodeGen.a  dependencies/ih264d/libih264d.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_adv-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_aui-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_propgrid-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_ribbon-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_stc-3.2.a  vcpkg_installed/x64-linux/lib/libwxscintilla-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_xrc-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_html-3.2.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_gl-3.2.a  /usr/lib/x86_64-linux-gnu/libGLU.so  /usr/lib/x86_64-linux-gnu/libGLX.so  /usr/lib/x86_64-linux-gnu/libEGL.so  /usr/lib/x86_64-linux-gnu/libOpenGL.so  -lffi  vcpkg_installed/x64-linux/lib/libwx_gtk3u_qa-3.2.a  vcpkg_installed/x64-linux/lib/libwx_baseu_xml-3.2.a  src/resource/libCemuResource.a  dependencies/ZArchive/libzarchive.a  vcpkg_installed/x64-linux/lib/libzstd.a  vcpkg_installed/x64-linux/lib/libzip.a  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libz.a  vcpkg_installed/x64-linux/lib/libcrypto.a  vcpkg_installed/x64-linux/lib/libpugixml.a  dependencies/cubeb/libcubeb.a  vcpkg_installed/x64-linux/lib/libboost_program_options.a  vcpkg_installed/x64-linux/lib/libboost_filesystem.a  vcpkg_installed/x64-linux/lib/libboost_nowide.a  vcpkg_installed/x64-linux/lib/libSDL2.a  -lwayland-egl  -lwayland-client  -lwayland-cursor  -lEGL  -lxkbcommon  /usr/lib/x86_64-linux-gnu/libXss.so  /usr/lib/x86_64-linux-gnu/libXxf86vm.so  -ldbus-1  -lunwind  -lunwind-generic  -lrt  -Wl,--no-undefined  vcpkg_installed/x64-linux/lib/libSDL2main.a  vcpkg_installed/x64-linux/lib/libwx_gtk3u_core-3.2.a  /usr/lib/x86_64-linux-gnu/libgtk-3.so  /usr/lib/x86_64-linux-gnu/libatk-bridge-2.0.so  /usr/lib/x86_64-linux-gnu/libatspi.so  vcpkg_installed/x64-linux/lib/libjpeg.a  vcpkg_installed/x64-linux/lib/libtiff.a  vcpkg_installed/x64-linux/lib/liblzma.a  vcpkg_installed/x64-linux/lib/libjpeg.a  vcpkg_installed/x64-linux/lib/libtiff.a  vcpkg_installed/x64-linux/lib/liblzma.a  /usr/lib/x86_64-linux-gnu/libXtst.so  /usr/lib/x86_64-linux-gnu/libdbus-1.so  /usr/lib/x86_64-linux-gnu/libpthread.so  /lib/x86_64-linux-gnu/libsystemd.so  /usr/lib/x86_64-linux-gnu/libgdk-3.so  /usr/lib/x86_64-linux-gnu/libXcomposite.so  /usr/lib/x86_64-linux-gnu/libXdamage.so  /usr/lib/x86_64-linux-gnu/libxkbcommon.so  /usr/lib/x86_64-linux-gnu/libwayland-cursor.so  /usr/lib/x86_64-linux-gnu/libwayland-egl.so  /usr/lib/x86_64-linux-gnu/libwayland-client.so  -lm  /usr/local/lib/x86_64-linux-gnu/libepoxy.so  /usr/lib/x86_64-linux-gnu/libdl.so  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libEGL.so  /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so  /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so  /usr/lib/x86_64-linux-gnu/libpango-1.0.so  /usr/lib/x86_64-linux-gnu/libfribidi.so  /usr/lib/x86_64-linux-gnu/libthai.so  /usr/lib/x86_64-linux-gnu/libdatrie.so  /usr/lib/x86_64-linux-gnu/libXft.so  /usr/lib/x86_64-linux-gnu/libharfbuzz.so  /usr/lib/x86_64-linux-gnu/libgraphite2.so  /usr/lib/x86_64-linux-gnu/libatk-1.0.so  /usr/lib/x86_64-linux-gnu/libcairo-gobject.so  /usr/lib/x86_64-linux-gnu/libcairo.so  /usr/lib/x86_64-linux-gnu/libpixman-1.so  /usr/lib/x86_64-linux-gnu/libfontconfig.so  /usr/lib/x86_64-linux-gnu/libuuid.so  /usr/lib/x86_64-linux-gnu/libfreetype.so  vcpkg_installed/x64-linux/lib/libpng16.a  /usr/lib/x86_64-linux-gnu/libxcb-shm.so  /usr/lib/x86_64-linux-gnu/libxcb-render.so  /usr/lib/x86_64-linux-gnu/libXrender.so  /usr/lib/x86_64-linux-gnu/libxcb.so  /usr/lib/x86_64-linux-gnu/libXau.so  /usr/lib/x86_64-linux-gnu/libXdmcp.so  /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so  /usr/lib/x86_64-linux-gnu/libgio-2.0.so  /usr/lib/x86_64-linux-gnu/libresolv.so  /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so  /usr/lib/x86_64-linux-gnu/libmount.so  /usr/lib/x86_64-linux-gnu/libblkid.so  /usr/lib/x86_64-linux-gnu/libselinux.so  /usr/lib/x86_64-linux-gnu/libsepol.so  /usr/lib/x86_64-linux-gnu/libpcre2-8.so  /usr/lib/x86_64-linux-gnu/libgobject-2.0.so  /usr/lib/x86_64-linux-gnu/libffi.so  /usr/lib/x86_64-linux-gnu/libglib-2.0.so  /usr/lib/x86_64-linux-gnu/libpcre.so  /usr/lib/x86_64-linux-gnu/libXtst.so  /usr/lib/x86_64-linux-gnu/libdbus-1.so  /usr/lib/x86_64-linux-gnu/libpthread.so  /lib/x86_64-linux-gnu/libsystemd.so  /usr/lib/x86_64-linux-gnu/libgdk-3.so  /usr/lib/x86_64-linux-gnu/libXcomposite.so  /usr/lib/x86_64-linux-gnu/libXdamage.so  /usr/lib/x86_64-linux-gnu/libxkbcommon.so  /usr/lib/x86_64-linux-gnu/libwayland-cursor.so  /usr/lib/x86_64-linux-gnu/libwayland-egl.so  /usr/lib/x86_64-linux-gnu/libwayland-client.so  -lm  /usr/local/lib/x86_64-linux-gnu/libepoxy.so  /usr/lib/x86_64-linux-gnu/libdl.so  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libEGL.so  /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so  /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so  /usr/lib/x86_64-linux-gnu/libpango-1.0.so  /usr/lib/x86_64-linux-gnu/libfribidi.so  /usr/lib/x86_64-linux-gnu/libthai.so  /usr/lib/x86_64-linux-gnu/libdatrie.so  /usr/lib/x86_64-linux-gnu/libXft.so  /usr/lib/x86_64-linux-gnu/libharfbuzz.so  /usr/lib/x86_64-linux-gnu/libgraphite2.so  /usr/lib/x86_64-linux-gnu/libatk-1.0.so  /usr/lib/x86_64-linux-gnu/libcairo-gobject.so  /usr/lib/x86_64-linux-gnu/libcairo.so  /usr/lib/x86_64-linux-gnu/libpixman-1.so  /usr/lib/x86_64-linux-gnu/libfontconfig.so  /usr/lib/x86_64-linux-gnu/libuuid.so  /usr/lib/x86_64-linux-gnu/libfreetype.so  vcpkg_installed/x64-linux/lib/libpng16.a  /usr/lib/x86_64-linux-gnu/libxcb-shm.so  /usr/lib/x86_64-linux-gnu/libxcb-render.so  /usr/lib/x86_64-linux-gnu/libXrender.so  /usr/lib/x86_64-linux-gnu/libxcb.so  /usr/lib/x86_64-linux-gnu/libXau.so  /usr/lib/x86_64-linux-gnu/libXdmcp.so  /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so  /usr/lib/x86_64-linux-gnu/libgio-2.0.so  /usr/lib/x86_64-linux-gnu/libresolv.so  /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so  /usr/lib/x86_64-linux-gnu/libmount.so  /usr/lib/x86_64-linux-gnu/libblkid.so  /usr/lib/x86_64-linux-gnu/libselinux.so  /usr/lib/x86_64-linux-gnu/libsepol.so  /usr/lib/x86_64-linux-gnu/libpcre2-8.so  /usr/lib/x86_64-linux-gnu/libgobject-2.0.so  /usr/lib/x86_64-linux-gnu/libffi.so  /usr/lib/x86_64-linux-gnu/libglib-2.0.so  /usr/lib/x86_64-linux-gnu/libpcre.so  /usr/lib/x86_64-linux-gnu/libXcursor.so  /usr/lib/x86_64-linux-gnu/libXinerama.so  /usr/lib/x86_64-linux-gnu/libXi.so  /usr/lib/x86_64-linux-gnu/libXrandr.so  /usr/local/lib/libXfixes.so  vcpkg_installed/x64-linux/lib/libexpat.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  -lm  vcpkg_installed/x64-linux/lib/libwx_baseu-3.2.a  -pthread  vcpkg_installed/x64-linux/lib/libz.a  -lc  -ldl  -lpthread && :
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandvideo.c.o): in function `Wayland_VideoInit':
SDL_waylandvideo.c:(.text+0xf7a): undefined reference to `libdecor_new'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandvideo.c.o): in function `Wayland_VideoQuit':
SDL_waylandvideo.c:(.text+0x1b18): undefined reference to `libdecor_unref'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `SetFullscreen':
SDL_waylandwindow.c:(.text+0x1541): undefined reference to `libdecor_frame_unset_fullscreen'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1555): undefined reference to `libdecor_frame_unset_capabilities'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x15d5): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x15e8): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1776): undefined reference to `libdecor_frame_set_capabilities'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `decoration_frame_configure':
SDL_waylandwindow.c:(.text+0x1bc5): undefined reference to `libdecor_configuration_get_window_state'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1c42): undefined reference to `libdecor_configuration_get_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1dc3): undefined reference to `libdecor_state_new'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1dd4): undefined reference to `libdecor_frame_commit'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1ddc): undefined reference to `libdecor_state_free'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1e88): undefined reference to `libdecor_configuration_get_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1ec1): undefined reference to `libdecor_frame_set_capabilities'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1f1d): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x1f2c): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2011): undefined reference to `libdecor_frame_unset_capabilities'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2051): undefined reference to `libdecor_configuration_get_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_GetWindowWMInfo':
SDL_waylandwindow.c:(.text+0x2173): undefined reference to `libdecor_frame_get_xdg_surface'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2266): undefined reference to `libdecor_frame_get_xdg_toplevel'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowModalFor':
SDL_waylandwindow.c:(.text+0x22dd): undefined reference to `libdecor_frame_set_parent'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_HideWindow':
SDL_waylandwindow.c:(.text+0x2403): undefined reference to `libdecor_frame_unref'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowFullscreen':
SDL_waylandwindow.c:(.text+0x27b4): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x27c8): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2869): undefined reference to `libdecor_frame_set_fullscreen'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2956): undefined reference to `libdecor_frame_set_capabilities'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2991): undefined reference to `libdecor_frame_unset_fullscreen'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x29aa): undefined reference to `libdecor_frame_unset_capabilities'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_RestoreWindow':
SDL_waylandwindow.c:(.text+0x29fc): undefined reference to `libdecor_frame_unset_maximized'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowResizable':
SDL_waylandwindow.c:(.text+0x2bd1): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_MaximizeWindow':
SDL_waylandwindow.c:(.text+0x2d4c): undefined reference to `libdecor_frame_set_maximized'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_MinimizeWindow':
SDL_waylandwindow.c:(.text+0x2de6): undefined reference to `libdecor_frame_set_minimized'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowMinimumSize':
SDL_waylandwindow.c:(.text+0x2f74): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowMaximumSize':
SDL_waylandwindow.c:(.text+0x3104): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowSize':
SDL_waylandwindow.c:(.text+0x340e): undefined reference to `libdecor_frame_is_floating'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x3462): undefined reference to `libdecor_state_new'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x3473): undefined reference to `libdecor_frame_commit'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x347b): undefined reference to `libdecor_state_free'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowTitle':
SDL_waylandwindow.c:(.text+0x352c): undefined reference to `libdecor_frame_set_title'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_ShowWindow':
SDL_waylandwindow.c:(.text+0x39a7): undefined reference to `libdecor_decorate'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x39c3): undefined reference to `libdecor_frame_set_app_id'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x39cc): undefined reference to `libdecor_frame_map'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x3a20): undefined reference to `libdecor_frame_set_visibility'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x3a7b): undefined reference to `libdecor_frame_set_visibility'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `SetFullscreen':
SDL_waylandwindow.c:(.text+0x1696): undefined reference to `libdecor_frame_set_fullscreen'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowBordered':
SDL_waylandwindow.c:(.text+0x2aa8): undefined reference to `libdecor_frame_set_visibility'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowResizable':
SDL_waylandwindow.c:(.text+0x2b83): undefined reference to `libdecor_frame_set_capabilities'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2bee): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: SDL_waylandwindow.c:(.text+0x2cef): undefined reference to `libdecor_frame_unset_capabilities'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowMinimumSize':
SDL_waylandwindow.c:(.text+0x2f91): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandwindow.c.o): in function `Wayland_SetWindowMaximumSize':
SDL_waylandwindow.c:(.text+0x3121): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylanddyn.c.o): in function `SDL_WAYLAND_LoadSymbols':
SDL_waylanddyn.c:(.text+0xa72): undefined reference to `libdecor_unref'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xa80): undefined reference to `libdecor_new'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xa8e): undefined reference to `libdecor_decorate'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xa9c): undefined reference to `libdecor_frame_unref'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xaaa): undefined reference to `libdecor_frame_set_title'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xab8): undefined reference to `libdecor_frame_set_app_id'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xac6): undefined reference to `libdecor_frame_set_max_content_size'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xad4): undefined reference to `libdecor_frame_set_min_content_size'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xae2): undefined reference to `libdecor_frame_resize'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xaf0): undefined reference to `libdecor_frame_move'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xafe): undefined reference to `libdecor_frame_commit'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb0c): undefined reference to `libdecor_frame_set_minimized'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb1a): undefined reference to `libdecor_frame_set_maximized'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb28): undefined reference to `libdecor_frame_unset_maximized'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb36): undefined reference to `libdecor_frame_set_fullscreen'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb44): undefined reference to `libdecor_frame_unset_fullscreen'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb52): undefined reference to `libdecor_frame_set_capabilities'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb60): undefined reference to `libdecor_frame_unset_capabilities'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb6e): undefined reference to `libdecor_frame_has_capability'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb7c): undefined reference to `libdecor_frame_set_visibility'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb8a): undefined reference to `libdecor_frame_is_visible'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xb98): undefined reference to `libdecor_frame_is_floating'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xba6): undefined reference to `libdecor_frame_set_parent'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbb4): undefined reference to `libdecor_frame_get_xdg_surface'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbc2): undefined reference to `libdecor_frame_get_xdg_toplevel'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbd0): undefined reference to `libdecor_frame_map'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbde): undefined reference to `libdecor_state_new'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbec): undefined reference to `libdecor_state_free'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xbfa): undefined reference to `libdecor_configuration_get_content_size'
/usr/bin/ld: SDL_waylanddyn.c:(.text+0xc08): undefined reference to `libdecor_configuration_get_window_state'
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libSDL2.a(SDL_waylandevents.c.o): in function `pointer_handle_button':
SDL_waylandevents.c:(.text+0x18de): undefined reference to `libdecor_frame_move'
/usr/bin/ld: SDL_waylandevents.c:(.text+0x19f2): undefined reference to `libdecor_frame_resize'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.

To check a few more items off of @Anuskuss's [list of necessities](https://github.com/cemu-project/Cemu/issues/1#issuecomment-1226610562) for an Arch PKGBUILD and to add to the discussion, here's the PKGBUILD I've been working on:

<details closed>
<summary>PKGBUILD 8/25 11:57</summary>

```sh
_pkgname=cemu
pkgname=$_pkgname-git
pkgver=2.0.a2abffd
pkgrel=1
pkgdesc='Software to emulate Wii U games and applications on PC (Linux native)'
arch=('i686' 'x86_64')
url='https://github.com/cemu-project/Cemu'
license=('MPL2')
provides=('cemu')
depends=('gtk3'
	 'libsecret'
	 'libgcrypt'
	 'systemd-libs'
	 'freeglut'
	 'nasm'
	 'vulkan-driver')
makedepends=('clang'
             'cmake'
             'git'
             'ninja'
             'llvm'
             'llvm-libs'
             'vulkan-headers'
             'zip'
             'unzip')
source=("$_pkgname::git+https://github.com/cemu-project/Cemu.git"
        'git+https://github.com/Exzap/ZArchive.git'
        'git+https://github.com/mozilla/cubeb.git'
        'git+https://github.com/microsoft/vcpkg.git'
        # cubeb dependencies
        'git+https://github.com/arsenm/sanitizers-cmake.git'
        'git+https://github.com/google/googletest.git'
        'git+https://github.com/KhronosGroup/Vulkan-Headers.git')
md5sums=('SKIP'
         'SKIP'
         'SKIP'
         'SKIP'
         'SKIP'
         'SKIP'
         'SKIP')

pkgver() {
    cd "$srcdir/$_pkgname"
    echo "$(git tag | sed 's/v//').$(git rev-parse --short HEAD)"
}

prepare() {
    cd "$srcdir/$_pkgname"

    for submodule in dependencies/{ZArchive,cubeb,vcpkg,Vulkan-Headers}; do
        git submodule init ${submodule}
        git config submodule.${submodule}.url "$srcdir/${submodule##*/}"
        git submodule update --init
    done

    cd "$srcdir/$_pkgname"/dependencies/cubeb

    for submodule in {cmake/sanitizers-cmake,googletest}; do
        git submodule init ${submodule}
        git config submodule.${submodule}.url "$srcdir/${submodule##*/}"
        git submodule update --init
    done
}

build() {
    cd "$srcdir/$_pkgname"
    mkdir -p build && cd build
    
    cmake .. -G Ninja \
      -DCMAKE_MAKE_PROGRAM=$(which ninja) \
      -DCMAKE_C_COMPILER=$(which clang) \
      -DCMAKE_CXX_COMPILER=$(which clang++) \
      -DCMAKE_CXX_STANDARD=20 \
      -DCMAKE_C_STANDARD=17 \
      -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON \
      -DCMAKE_BUILD_TYPE=release \
      -DPUBLIC_RELEASE=ON \
      -DCMAKE_INSTALL_PREFIX=/usr
    ninja
}

package() {
    cd "$srcdir/$_pkgname/build"
    DESTDIR="$pkgdir" ninja install
}
```

</details>

We're getting very close to compiling the release build successfully on Clang 14, like @vncfrrll, it fails on the final linking step with an undefined reference to `MD5_Init`. I have a feeling this can be worked around with some linking flags which I'll look into later.

As for the current status, it's looking like this:
- [x] Determine depends and makedepends
- [x] Check for alternatives (clang=12 -> clang>=12, clang -> gcc, ninja -> make)
- [x] Submodules as source
- [x] Write pkgver
- [x] Remove patches
- [x] ~~Actually fix /usr/include/c++/12.2.0/bits/uniform_int_dist.h-related issue~~ No longer necessary
- [x] Don't delete CFLAGS/CXXFLAGS (was neccessary because of -Werror=format-security in my /etc/makepkg.conf)
- [ ] Don't build debug build (almost there)
- [ ] Strip binary
- [ ] Install (probably in /opt/cemu)
- [ ] Create .desktop file

The depends and makedepends were tested in a clean Arch chroot with [devtest](https://github.com/bscubed/devtest), a package I created for testing dependencies on Arch packages which you guys are also free to use while developing this PKGBUILD.

The pkgver is also WIP, right now it outputs something like `2.0.a2abffd` which seems fine to me, but I'm open to suggestions.

I might throw together a .desktop file next and get that approved by the Cemu dev team to make sure the information there is along the lines of what they're looking for.

> I'm able to compile on Arch Linux with these steps:
> 
> ```shell
> sudo pacman -S base-devel vulkan-headers gtk3 libsecret libgcrypt systemd-libs freeglut nasm clang llvm llvm-libs zip unzip make cmake
> export VULKAN_SDK=/usr
> git clone --recursive https://github.com/Cemu-project/Cemu
> cd Cemu
> sed -i 's:project(CemuCommon):project(CemuCommon CXX):g' src/Common/CMakeLists.txt # to get rid of the static_assert error
> mkdir build
> cd build
> cmake .. \
>     -DCMAKE_BUILD_TYPE=debug \
>     -DCMAKE_C_COMPILER=$(which clang) \
>     -DCMAKE_CXX_COMPILER=$(which clang++) \
>     -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON \ # I had an issue with RPATH, with this flag it disappeared
>     -DCMAKE_CXX_STANDARD=20 \
>     -DCMAKE_C_STANDARD=17 \
>     -DPUBLIC_RELEASE=ON
> make -j$(nproc)
> ```
> 
> I'm almost there with a working PKGBUILD:
> 
> ```shell
> pkgname=cemu
> pkgver=2.0
> pkgrel=1
> pkgdesc='Cemu is a Wii U emulator'
> arch=('any')
> license=('MPL2')
> makedepends=(vulkan-headers cmake gtk3 libsecret libgcrypt systemd-libs freeglut nasm clang llvm llvm-libs zip unzip make)
> url='https://github.com/cemu-project/Cemu'
> source=("git+$url#commit=be0f062de82737639a0c16cf59c0d0e2c08f41be")
> sha256sums=('SKIP')
> 
> prepare(){
>   cd Cemu
>   git submodule update --init --recursive
>   # needed to get rid of the static_assert error
>   sed -i 's:project(CemuCommon):project(CemuCommon CXX):g' src/Common/CMakeLists.txt
>   mkdir -p build
>   export VULKAN_SDK=/usr
> }
> 
> build() {
>   cd Cemu/build
>   cmake .. \
>       -DCMAKE_BUILD_TYPE=debug \
>       -DCMAKE_MAKE_PROGRAM=$(which make) \ # otherwise it throws an error asking to set this flag
>       -DCMAKE_C_COMPILER=$(which clang) \
>       -DCMAKE_CXX_COMPILER=$(which clang++) \
>       -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON \
>       -DCMAKE_CXX_STANDARD=20 \
>       -DCMAKE_C_STANDARD=17 \
>       -DCMAKE_INSTALL_PREFIX=/usr # at the moment this flag is useless, maybe we should install it in /opt/Cemu
>   make
> }
> 
> package() {
>   cd Cemu
>   install -Dm644 LICENSE.txt "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
>   # install the package, possibly in /opt/cemu, needs write permissions to its own directory to start
> }
> ```

cmake fails with the following error:
```
/home/user/cemu-git/PKGBUILD: line 27: -DCMAKE_C_COMPILER=/usr/bin/clang: No such file or directory
==> ERROR: A failure occurred in build().
    Aborting...
```

Had to add a second backslash to `-DCMAKE_MAKE_PROGRAM=$(which make) \ # otherwise it throws an error asking to set this flag` for it to work.

After fixing this issue I'm getting another error:
```
[100%] Linking CXX executable Cemu
/usr/bin/ld: cannot open output file Cemu: Is a directory
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/CMakeFiles/CemuBin.dir/build.make:367: src/Cemu] Error 1
make[1]: *** [CMakeFiles/Makefile2:554: src/CMakeFiles/CemuBin.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

I think it's because I'm building on a case-sensitive filesystem (ext4 with default options)

> @bbaster Everytime that particular error pops up, you will almost certainly be hit with MD5 linker errors afterwards. Try my [PKGBUILD](https://github.com/cemu-project/Cemu/issues/1#issuecomment-1226610562) instead.
> 
> **Edit:** Just saw your comment in the other thread that my PKGBUILD gives you the same error. Then you're sadly not following directions because it has worked for others (on case-sensitive filesystems). I'm also not sure what's triggering it. Anyways, try [this](https://github.com/cemu-project/Cemu/issues/1#issuecomment-1227343291).

@Anuskuss I am following directions, I were using an older version of your PKGBUILD. Now after using an almost unchanged PKGBUILD (just had to replace `sudo pacman -U...` with `sudo pacman -Udd...` since both KDevelop and Qt Creator both explicitly require the latest version of Clang) I'm getting the same "undefined reference to \`libdecor..." errors, interestingly I didn't see anything related to MD5, maybe I will once the libdecor issue is resolved.

Also could this PKGBUILD use the latest version of Clang, not 12? Rebooting after downgrading these packages makes GPU acceleration completely broken (at least with Intel/AMD GPUs, I think it could work on NVIDIA as it doesn't use Mesa which depends on `llvm-libs`). After changing your PKGBUILD to not downgrade Clang and LLVM and updating them I got the same error, I don't recall seeing anyone in this thread getting an error with Clang 14 but not with Clang 12 either.

> > @bbaster Everytime that particular error pops up, you will almost certainly be hit with MD5 linker errors afterwards. Try my [PKGBUILD](https://github.com/cemu-project/Cemu/issues/1#issuecomment-1226610562) instead.
> > **Edit:** Just saw your comment in the other thread that my PKGBUILD gives you the same error. Then you're sadly not following directions because it has worked for others (on case-sensitive filesystems). I'm also not sure what's triggering it. Anyways, try [this](https://github.com/cemu-project/Cemu/issues/1#issuecomment-1227343291).
> 
> @Anuskuss I am following directions, I were using an older version of your PKGBUILD. Now after using an almost unchanged PKGBUILD (just had to replace `sudo pacman -U...` with `sudo pacman -Udd...` since both KDevelop and Qt Creator both explicitly require the latest version of Clang) I'm getting the same "undefined reference to `libdecor..." errors, interestingly I didn't see anything related to MD5, maybe I will once the libdecor issue is resolved.
> 
> Also could this PKGBUILD use the latest version of Clang, not 12? Rebooting after downgrading these packages makes GPU acceleration completely broken (at least with Intel/AMD GPUs, I think it could work on NVIDIA as it doesn't use Mesa which depends on `llvm-libs`). After changing your PKGBUILD to not downgrade Clang and LLVM and updating them I got the same error, I don't recall seeing anyone in this thread getting an error with Clang 14 but not with Clang 12 either.

This might sound obvious, but I checked, and the "`libdecor`" linker failure only occurs while Arch's `libdecor` from `Extra` is installed in the system. Maybe this is one for the `conflicts` for the time being (though I look forward to seeing `libdecor` used correctly alongside all the other dependencies that will at some point be migrated to stable, current, signed, precompiled, packaged, shared builds). 

When I did that outside of clean chroots, I **did** have to **delete a folder that I noticed `vcpkg` has been creating, `~/.cache/vcpkg`, before the effects of uninstalling `libdecor` would "apply" to the build process**. This further justifies my use of clean chroots, and I admit that until this point, I thought the revulsion for `vcpkg` was a little overblown. I hadn't realized that it literally starts dumping files into a huge persistent cache that then breaks. I think I've started to see the problem with it now. 200+ MB in the home directory, when ideally all intermediate build artifacts from `PKGBUILD`s should stay in the `PKGBUILD` folder.

> @abouvier
> 
> > The final linker error is caused by [these lines](https://github.com/cemu-project/Cemu/blob/ca78b927183eeefa91c938525318ba8eb07d376a/src/Cemu/Logging/CemuLogging.h#L61-L63). The project compiles fine in release mode if they are commented.
> 
> What error are you referring to? Is this related to #74?

This one:
```
/usr/bin/ld: CMakeFiles/CemuBin.dir/main.cpp.o: in function `std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> > fmt::v7::detail::vformat<char8_t, 0>(fmt::v7::basic_string_view<char8_t>, fmt::v7::basic_format_args<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<fmt::v7::type_identity<char8_t>::type>, fmt::v7::type_identity<char8_t>::type> >)':
main.cpp:(.text._ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE[_ZN3fmt2v76detail7vformatIDuLi0EEENSt7__cxx1112basic_stringIT_St11char_traitsIS5_ESaIS5_EEENS0_17basic_string_viewIS5_EENS0_17basic_format_argsINS0_20basic_format_contextINS1_15buffer_appenderINS0_13type_identityIS5_E4typeEEESH_EEEE]+0x6f): undefined reference to `void std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::_M_construct<char8_t const*>(char8_t const*, char8_t const*, std::forward_iterator_tag)'
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```
I don't think it's related, because this error should be fixed with fmt 7.1.3 too.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,9 @@ endif()
 
 
 project(Cemu VERSION 0.1)
+
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
@@ -64,6 +67,8 @@ option(ENABLE_CUBEB "Enabled cubeb backend" ON)
 
 option(ENABLE_WXWIDGETS "Build with wxWidgets UI (Currently required)" ON)
 
+set(THREADS_PREFER_PTHREAD_FLAG true)
+find_package(Threads REQUIRED)
 find_package(SDL2 REQUIRED)
 find_package(CURL REQUIRED)
 find_package(pugixml REQUIRED)
@@ -73,7 +78,20 @@ find_package(Boost COMPONENTS program_options filesystem nowide REQUIRED)
 find_package(libzip REQUIRED)
 find_package(glslang REQUIRED)
 find_package(ZLIB REQUIRED)
-find_package(zstd REQUIRED)
+find_package(zstd MODULE REQUIRED) # MODULE so that zstd::zstd is available
+find_package(OpenSSL COMPONENTS Crypto SSL REQUIRED)
+find_package(glm REQUIRED)
+find_package(fmt 7.0.0 REQUIRED)
+find_package(PNG REQUIRED)
+
+# glslang versions older than 11.11.0 define targets without a namespace
+if (NOT TARGET glslang::SPIRV AND TARGET SPIRV)
+	add_library(glslang::SPIRV ALIAS SPIRV)
+endif()
+
+if (UNIX)
+	find_package(X11 REQUIRED)
+endif()
 
 if (ENABLE_VULKAN)
   include_directories("dependencies/Vulkan-Headers/include")
@@ -93,29 +111,25 @@ if (ENABLE_WXWIDGETS)
 	find_package(wxWidgets 3.2 REQUIRED COMPONENTS base core gl propgrid xrc)
 endif()
 
-find_package(OpenSSL REQUIRED)
-find_package(X11)
-
-# find a better way to handle this
-link_libraries(${Boost_LIBRARIES})
-link_libraries(${X11_LIBRARIES})
-link_libraries(SDL2::SDL2 SDL2::SDL2main SDL2::SDL2-static)
-if (ENABLE_WXWIDGETS)
-  link_libraries(wx::core wx::base)
-endif()
-
 if (ENABLE_CUBEB)
-  option(BUILD_TESTS "" OFF)
-  option(BUILD_TOOLS "" OFF)
-  option(BUNDLE_SPEEX "" OFF)
-  set(USE_WINMM OFF CACHE BOOL "")
-  add_subdirectory(dependencies/cubeb)
-  set_property(TARGET cubeb PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-  link_libraries(cubeb)
-  add_compile_definitions(HAS_CUBEB=1)
+	find_package(cubeb)
+	if (NOT cubeb_FOUND)
+		option(BUILD_TESTS "" OFF)
+		option(BUILD_TOOLS "" OFF)
+		option(BUNDLE_SPEEX "" OFF)
+		set(USE_WINMM OFF CACHE BOOL "")
+		add_subdirectory("dependencies/cubeb")
+		set_property(TARGET cubeb PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+		add_library(cubeb::cubeb ALIAS cubeb)
+	endif()
+	add_compile_definitions("HAS_CUBEB=1")
 endif()
 
-add_subdirectory(dependencies/ih264d)
-add_subdirectory(dependencies/ZArchive)
+add_subdirectory("dependencies/ih264d")
+
+find_package(ZArchive)
+if (NOT ZArchive_FOUND)
+	add_subdirectory("dependencies/ZArchive")
+endif()
 
 add_subdirectory(src)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,19 +1,25 @@
 cmake_minimum_required(VERSION 3.21.1)
 
 option(PUBLIC_RELEASE "Compile with debug asserts disabled and no console" OFF)
+option(ENABLE_VCPKG "Enable the vcpkg package manager" ON)
 
 if (PUBLIC_RELEASE)
   add_definitions(-DPUBLIC_RELEASE)
   set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE) # enable LTO
 endif()
 
-if (WIN32)
-  set(VCPKG_TARGET_TRIPLET "x64-windows-static" CACHE STRING "")
+if (ENABLE_VCPKG)
+	set(VCPKG_OVERLAY_PORTS "${CMAKE_CURRENT_LIST_DIR}/dependencies/vcpkg_overlay_ports")
+	set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/dependencies/vcpkg/scripts/buildsystems/vcpkg.cmake"
+		CACHE STRING "Vcpkg toolchain file")
+	# Set this so that all the various find_package() calls don't need an explicit
+	# CONFIG option
+	set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
+	if (WIN32)
+		set(VCPKG_TARGET_TRIPLET "x64-windows-static" CACHE STRING "")
+	endif()
 endif()
 
-set(VCPKG_OVERLAY_PORTS "${CMAKE_CURRENT_LIST_DIR}/dependencies/vcpkg_overlay_ports")
-set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/dependencies/vcpkg/scripts/buildsystems/vcpkg.cmake"
-  CACHE STRING "Vcpkg toolchain file")
 
 project(Cemu VERSION 0.1)
 set(CMAKE_CXX_STANDARD 20)
@@ -58,16 +64,16 @@ option(ENABLE_CUBEB "Enabled cubeb backend" ON)
 
 option(ENABLE_WXWIDGETS "Build with wxWidgets UI (Currently required)" ON)
 
-find_package(SDL2 CONFIG REQUIRED)
-find_package(CURL CONFIG REQUIRED)
-find_package(pugixml CONFIG REQUIRED)
-find_package(imgui CONFIG REQUIRED)
-find_package(RapidJSON CONFIG REQUIRED)
+find_package(SDL2 REQUIRED)
+find_package(CURL REQUIRED)
+find_package(pugixml REQUIRED)
+find_package(imgui REQUIRED)
+find_package(RapidJSON REQUIRED)
 find_package(Boost COMPONENTS program_options filesystem nowide REQUIRED)
 find_package(libzip REQUIRED)
 find_package(glslang REQUIRED)
 find_package(ZLIB REQUIRED)
-find_package(zstd CONFIG REQUIRED)
+find_package(zstd REQUIRED)
 
 if (ENABLE_VULKAN)
   include_directories("dependencies/Vulkan-Headers/include")
@@ -84,7 +90,7 @@ if (ENABLE_DISCORD_RPC)
 endif()
 
 if (ENABLE_WXWIDGETS)
-  find_package(wxWidgets CONFIG REQUIRED)
+	find_package(wxWidgets 3.2 REQUIRED COMPONENTS base core gl propgrid xrc)
 endif()
 
 find_package(OpenSSL REQUIRED)

--- a/cmake/FindZArchive.cmake
+++ b/cmake/FindZArchive.cmake
@@ -1,0 +1,20 @@
+# SPDX-FileCopyrightText: 2022 Andrea Pappacoda <andrea@pappacoda.it>
+# SPDX-License-Identifier: ISC
+
+find_package(PkgConfig)
+
+if (PKG_CONFIG_FOUND)
+	pkg_search_module(zarchive IMPORTED_TARGET GLOBAL zarchive)
+	if (zarchive_FOUND)
+		add_library(ZArchive::zarchive ALIAS PkgConfig::zarchive)
+	endif()
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(ZArchive
+	REQUIRED_VARS
+		zarchive_LINK_LIBRARIES
+		zarchive_FOUND
+	VERSION_VAR
+		zarchive_VERSION
+)

--- a/cmake/Findimgui.cmake
+++ b/cmake/Findimgui.cmake
@@ -1,0 +1,27 @@
+# SPDX-FileCopyrightText: 2022 Andrea Pappacoda <andrea@pappacoda.it>
+# SPDX-License-Identifier: ISC
+
+include(FindPackageHandleStandardArgs)
+
+find_package(imgui CONFIG)
+if (imgui_FOUND)
+	# Use upstream imguiConfig.cmake if possible
+	find_package_handle_standard_args(imgui CONFIG_MODE)
+else()
+	# Fallback to pkg-config otherwise
+	find_package(PkgConfig)
+	if (PKG_CONFIG_FOUND)
+		pkg_search_module(imgui IMPORTED_TARGET GLOBAL imgui)
+		if (imgui_FOUND)
+			add_library(imgui::imgui ALIAS PkgConfig::imgui)
+		endif()
+	endif()
+
+	find_package_handle_standard_args(imgui
+		REQUIRED_VARS
+			imgui_LINK_LIBRARIES
+			imgui_FOUND
+		VERSION_VAR
+			imgui_VERSION
+	)
+endif()

--- a/cmake/FindwxWidgets.cmake
+++ b/cmake/FindwxWidgets.cmake
@@ -1,0 +1,49 @@
+# SPDX-FileCopyrightText: 2022 Andrea Pappacoda <andrea@pappacoda.it>
+# SPDX-License-Identifier: ISC
+
+include(FindPackageHandleStandardArgs)
+find_package(wxWidgets CONFIG COMPONENTS ${wxWidgets_FIND_COMPONENTS})
+
+if (wxWidgets_FOUND)
+	# Use upstream wxWidgetsConfig.cmake if possible
+	find_package_handle_standard_args(wxWidgets CONFIG_MODE)
+else()
+	# Fall back to CMake's FindwxWidgets
+	# Temporarily unset CMAKE_MODULE_PATH to avoid calling the current find
+	# module recursively
+	set(_tmp_module_path "${CMAKE_MODULE_PATH}")
+	set(CMAKE_MODULE_PATH "")
+
+	find_package(wxWidgets MODULE QUIET COMPONENTS ${wxWidgets_FIND_COMPONENTS})
+
+	set(CMAKE_MODULE_PATH "${_tmp_module_path}")
+	unset(_tmp_module_path)
+
+	if (wxWidgets_FOUND)
+		add_library(wx::base IMPORTED INTERFACE)
+		target_include_directories(wx::base INTERFACE ${wxWidgets_INCLUDE_DIRS})
+		target_link_libraries(wx::base INTERFACE ${wxWidgets_LIBRARIES})
+		target_link_directories(wx::base INTERFACE ${wxWidgets_LIBRARY_DIRS})
+		target_compile_definitions(wx::base INTERFACE ${wxWidgets_DEFINITIONS})
+		target_compile_options(wx::base INTERFACE ${wxWidgets_CXX_FLAGS})
+
+		# FindwxWidgets sets everything into a single set of variables, so it is
+		# impossible to tell what libraries are required for what component.
+		# To be compatible with wxWidgetsConfig, we create an alias for each
+		# component so that the user can still use target_link_libraries(wx::gl)
+		foreach(component ${wxWidgets_FIND_COMPONENTS})
+			if (NOT component STREQUAL "base")
+				# don't alias base to itself
+				add_library(wx::${component} ALIAS wx::base)
+			endif()
+		endforeach()
+	endif()
+
+	find_package_handle_standard_args(wxWidgets
+		REQUIRED_VARS
+			wxWidgets_LIBRARIES
+			wxWidgets_FOUND
+		VERSION_VAR
+			wxWidgets_VERSION_STRING
+	)
+endif()

--- a/cmake/Findzstd.cmake
+++ b/cmake/Findzstd.cmake
@@ -1,0 +1,33 @@
+# SPDX-FileCopyrightText: 2022 Andrea Pappacoda <andrea@pappacoda.it>
+# SPDX-License-Identifier: ISC
+
+include(FindPackageHandleStandardArgs)
+
+find_package(zstd CONFIG)
+if (zstd_FOUND)
+	# Use upstream zstdConfig.cmake if possible
+	if (NOT TARGET zstd::zstd)
+		if (TARGET zstd::libzstd_static)
+			add_library(zstd::zstd ALIAS zstd::libzstd_static)
+		elseif (TARGET zstd::libzstd_shared)
+			add_library(zstd::zstd ALIAS zstd::libzstd_shared)
+		endif()
+	endif()
+	find_package_handle_standard_args(zstd CONFIG_MODE)
+else()
+	# Fallback to pkg-config otherwise
+	find_package(PkgConfig)
+	if (PKG_CONFIG_FOUND)
+		pkg_search_module(libzstd IMPORTED_TARGET GLOBAL libzstd)
+		if (libzstd_FOUND)
+			add_library(zstd::zstd ALIAS PkgConfig::libzstd)
+		endif()
+	endif()
+
+	find_package_handle_standard_args(zstd
+		REQUIRED_VARS
+			libzstd_LINK_LIBRARIES
+			libzstd_FOUND
+		VERSION_VAR libzstd_VERSION
+	)
+endif()

--- a/dependencies/discord-rpc/src/CMakeLists.txt
+++ b/dependencies/discord-rpc/src/CMakeLists.txt
@@ -98,8 +98,7 @@ if(UNIX)
     endif (APPLE)
 endif(UNIX)
 
-target_link_libraries(discord-rpc PRIVATE rapidjson)
-#target_include_directories(discord-rpc PRIVATE ${RAPIDJSON}/include)
+target_include_directories(discord-rpc PRIVATE ${RAPIDJSON_INCLUDE_DIRS})
 
 if (NOT ${ENABLE_IO_THREAD})
     target_compile_definitions(discord-rpc PUBLIC -DDISCORD_DISABLE_IO_THREAD)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -79,18 +79,21 @@ set_target_properties(CemuBin PROPERTIES
         OUTPUT_NAME "Cemu"
         )
 
-target_link_libraries(CemuBin PRIVATE CemuCommon CemuComponents CemuCafe CemuConfig CemuGui CemuAudio CemuInput CemuUtil)
+target_link_libraries(CemuBin PRIVATE
+	CemuAudio
+	CemuCafe
+	CemuCommon
+	CemuComponents
+	CemuConfig
+	CemuGui
+	CemuInput
+	CemuUtil
+)
+
 target_link_libraries(CemuBin PRIVATE CemuAsm)
-target_link_libraries(CemuBin PRIVATE OpenSSL::SSL)
-target_link_libraries(CemuBin PRIVATE ZLIB::ZLIB)
-target_link_libraries(CemuBin PRIVATE ${wxWidgets_LIBRARIES})
-target_link_libraries(CemuBin PRIVATE CURL::libcurl)
-target_link_libraries(CemuBin PRIVATE imgui::imgui)
-target_link_libraries(CemuBin PRIVATE pugixml pugixml::static pugixml::pugixml)
+target_link_libraries(CemuBin PRIVATE SDL2::SDL2 SDL2::SDL2main) # is SDL2main needed?
+target_link_libraries(CemuBin PRIVATE imguiImpl OpenGL::GL)
 
-target_link_libraries(CemuBin PUBLIC
-CemuCommon CemuAudio CemuInput CemuComponents CemuCafe CemuConfig CemuGui imguiImpl)
-
-# needed because of some cyclic dependencies. fix this
-target_link_libraries(CemuBin PUBLIC
-CemuCommon CemuInput CemuComponents CemuCafe CemuResource CemuGui CemuAsm)
+if (ENABLE_WXWIDGETS)
+	target_link_libraries(CemuBin PRIVATE wx::base wx::core)
+endif()

--- a/src/Cafe/CMakeLists.txt
+++ b/src/Cafe/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(CemuCafe)
 
-include_directories(".")
-
 if((CMAKE_C_COMPILER_ID STREQUAL "GNU") OR (CMAKE_C_COMPILER_ID STREQUAL "Clang"))
 	add_compile_options(-mssse3 -mavx2)
 endif()
@@ -14,20 +12,39 @@ set_property(TARGET CemuCafe PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CON
 
 target_precompile_headers(CemuCafe PRIVATE ../Common/precompiled.h)
 
-target_include_directories(CemuCafe PRIVATE ../)
+target_include_directories(CemuCafe PUBLIC "../")
 
-#target_link_libraries(CemuCafe ZArchivexx)
-#target_link_libraries(CemuCafe CemuCommon CemuCore CemuConfig CemuUtil CemuResource)
-#target_link_libraries(CemuCafe OpenSSL::SSL)
-#target_link_libraries(CemuCafe ZLIB::ZLIB)
-#target_link_libraries(CemuCafe imgui::imgui)
-#target_link_libraries(CemuCafe imguiImpl)
-#target_link_libraries(CemuCafe pugixml pugixml::static pugixml::pugixml)
-#target_link_libraries(CemuCafe libzip::zip)
-target_link_libraries(CemuCafe glslang SPIRV)
-target_link_libraries(CemuCafe ih264d zarchive)
-#target_link_libraries(CemuCafe zstd::libzstd_static)
+target_link_libraries(CemuCafe PRIVATE
+	CemuAsm
+	CemuAudio
+	CemuCommon
+	CemuComponents
+	CemuConfig
+	CemuGui
+	CemuInput
+	CemuResource
+	CemuUtil
+	imguiImpl
+	Boost::headers
+	Boost::nowide
+	CURL::libcurl
+	fmt::fmt
+	glslang::SPIRV
+	ih264d
+	imgui::imgui
+	OpenSSL::Crypto
+	OpenSSL::SSL
+	PNG::PNG
+	pugixml::pugixml
+	ZArchive::zarchive
+	ZLIB::ZLIB
+	zstd::zstd
+)
 
-IF(WIN32)
-target_link_libraries(CemuCafe iphlpapi)
-ENDIF()
+if (ENABLE_WXWIDGETS)
+	target_link_libraries(CemuCafe PRIVATE wx::base wx::core)
+endif()
+
+if(WIN32)
+	target_link_libraries(CemuCafe PRIVATE iphlpapi)
+endif()

--- a/src/Cemu/CMakeLists.txt
+++ b/src/Cemu/CMakeLists.txt
@@ -8,9 +8,25 @@ set_property(TARGET CemuComponents PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$
 
 target_precompile_headers(CemuComponents PRIVATE ../Common/precompiled.h)
 
-target_include_directories(CemuComponents PRIVATE ../)
+target_include_directories(CemuComponents PUBLIC "../")
+
+target_link_libraries(CemuComponents PRIVATE
+	CemuCafe
+	CemuCommon
+	CemuConfig
+	CemuGui
+	CemuUtil
+	Boost::headers
+	CURL::libcurl
+	OpenSSL::Crypto
+	OpenSSL::SSL
+	pugixml::pugixml
+	ZLIB::ZLIB
+)
+
+# PUBLIC because fmt/format.h is included in ExpressionParser/ExpressionParser.h
+target_link_libraries(CemuComponents PUBLIC fmt::fmt)
 
 if(ENABLE_DISCORD_RPC)
 target_link_libraries(CemuComponents PRIVATE discord-rpc)
 endif()
-target_link_libraries(CemuComponents PRIVATE CemuUtil)

--- a/src/Common/CMakeLists.txt
+++ b/src/Common/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(CemuCommon)
 
-#include_directories(".")
-
 file(GLOB CPP_FILES *.cpp)
 file(GLOB H_FILES *.h)
 add_library(CemuCommon ${CPP_FILES} ${H_FILES})
@@ -29,4 +27,22 @@ target_sources(CemuCommon
         )
 
 target_precompile_headers(CemuCommon PUBLIC precompiled.h)
-target_include_directories(CemuCommon PRIVATE ../)
+target_include_directories(CemuCommon PUBLIC "../")
+
+target_link_libraries(CemuCommon PRIVATE
+	CemuCafe
+	CemuConfig
+	CemuComponents
+	Boost::nowide
+	Boost::filesystem
+	glm::glm
+)
+
+if (UNIX)
+	target_link_libraries(CemuCommon PRIVATE X11::X11 X11::Xrender X11::Xutil)
+endif()
+
+# PUBLIC because:
+# - boost/predef/os.h is included in platform.h
+# - fmt/core.h is included in precompiled.h
+target_link_libraries(CemuCommon PUBLIC Boost::headers fmt::fmt)

--- a/src/Common/platform.h
+++ b/src/Common/platform.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include "boost/predef/os.h"
+#include <boost/predef/os.h>
 #include <cstdint>
 
 #if BOOST_OS_WINDOWS

--- a/src/audio/CMakeLists.txt
+++ b/src/audio/CMakeLists.txt
@@ -34,9 +34,16 @@ endif()
 
 target_precompile_headers(CemuAudio PRIVATE ../Common/precompiled.h)
 
-#target_link_libraries(CemuAudio CemuCommon CemuGui CemuPlatform)
-target_include_directories(CemuAudio PRIVATE ../)
+target_include_directories(CemuAudio PUBLIC "../")
+
+target_link_libraries(CemuAudio PRIVATE CemuCafe CemuConfig CemuGui CemuUtil)
 
 if(ENABLE_CUBEB)
-  target_link_libraries(CemuAudio cubeb)
+	# PUBLIC because cubeb.h/cubeb.h is included in CubebAPI.h
+	target_link_libraries(CemuAudio PUBLIC cubeb::cubeb)
+endif()
+
+if (ENABLE_WXWIDGETS)
+	# PUBLIC because wx/wx.h is included in audioDebuggerWindow.h
+	target_link_libraries(CemuAudio PUBLIC wx::base wx::core)
 endif()

--- a/src/config/CMakeLists.txt
+++ b/src/config/CMakeLists.txt
@@ -8,4 +8,19 @@ set_property(TARGET CemuConfig PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<C
 
 target_precompile_headers(CemuConfig PRIVATE ../Common/precompiled.h)
 
-target_include_directories(CemuConfig PRIVATE ../)
+target_include_directories(CemuConfig PUBLIC "../")
+
+target_link_libraries(CemuConfig PRIVATE
+	CemuCafe
+	CemuCommon
+	CemuUtil
+	Boost::headers
+	Boost::program_options
+	pugixml::pugixml
+)
+
+if (ENABLE_WXWIDGETS)
+	# PUBLIC because wx/language.h is included in CemuConfig.h
+	# Could be made PRIVATE by using 0 instead of wxLANGUAGE_DEFAULT
+	target_link_libraries(CemuConfig PUBLIC wx::base wx::core)
+endif()

--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(CemuGui)
 
-include_directories(".")
-
 file(GLOB_RECURSE CPP_FILES *.cpp)
 file(GLOB_RECURSE H_FILES *.h)
 add_library(CemuGui ${CPP_FILES} ${H_FILES})
@@ -17,13 +15,32 @@ wxcomponents/checktree.h
 
 target_precompile_headers(CemuGui PRIVATE ../Common/precompiled.h)
 
-target_include_directories(CemuGui PRIVATE ../)
+target_include_directories(CemuGui PUBLIC "../")
+# PUBLIC because rapidjson/document.h is included in ChecksumTool.h
+target_include_directories(CemuGui PUBLIC ${RAPIDJSON_INCLUDE_DIRS})
 
-target_link_libraries(CemuGui ${wxWidgets_LIBRARIES})
-target_link_libraries(CemuGui libzip::zip)
-target_link_libraries(CemuGui CemuCafe CemuComponents CemuResource)
-target_link_libraries(CemuGui zarchive)
+target_link_libraries(CemuGui PRIVATE
+	CemuAudio
+	CemuCafe
+	CemuCommon
+	CemuComponents
+	CemuConfig
+	CemuInput
+	CemuResource
+	CemuUtil
+	Boost::headers
+	CURL::libcurl
+	libzip::zip
+	OpenSSL::Crypto
+	pugixml::pugixml
+	ZArchive::zarchive
+)
 
 if(ENABLE_CUBEB)
-    target_link_libraries(CemuGui cubeb)
+	target_link_libraries(CemuGui PRIVATE cubeb::cubeb)
+endif()
+
+if (ENABLE_WXWIDGETS)
+	# PUBLIC because wx/app.h is included in CemuApp.h
+	target_link_libraries(CemuGui PUBLIC wx::base wx::core wx::gl wx::propgrid wx::xrc)
 endif()

--- a/src/imgui/CMakeLists.txt
+++ b/src/imgui/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(imguiImpl)
 
-include_directories(".")
-
 add_library(imguiImpl)
 
 set_property(TARGET imguiImpl PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
@@ -17,5 +15,14 @@ target_sources(imguiImpl PRIVATE
 
 target_precompile_headers(imguiImpl PRIVATE ../Common/precompiled.h)
 
-target_link_libraries(CemuCommon)
-target_include_directories(imguiImpl PRIVATE ../)
+target_include_directories(imguiImpl PUBLIC "../")
+
+target_link_libraries(imguiImpl PRIVATE
+	CemuCafe
+	CemuCommon
+	CemuGui
+	CemuInput
+	CemuResource
+	CemuUtil
+	imgui::imgui
+)

--- a/src/input/CMakeLists.txt
+++ b/src/input/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(CemuInput)
 
-include_directories(".")
-
 add_library(CemuInput
 InputManager.cpp
 InputManager.h
@@ -95,4 +93,21 @@ endif()
 
 target_precompile_headers(CemuInput PRIVATE ../Common/precompiled.h)
 
-target_include_directories(CemuInput PRIVATE ../)
+target_include_directories(CemuInput PUBLIC "../")
+
+target_link_libraries(CemuInput PRIVATE
+	CemuCafe
+	CemuCommon
+	CemuConfig
+	CemuGui
+	CemuUtil
+	Boost::headers
+	Boost::program_options
+	glm::glm
+	pugixml::pugixml
+	SDL2::SDL2
+)
+
+if (ENABLE_WXWIDGETS)
+	target_link_libraries(CemuInput PRIVATE wx::base wx::core)
+endif()

--- a/src/resource/CMakeLists.txt
+++ b/src/resource/CMakeLists.txt
@@ -16,4 +16,6 @@ target_sources(CemuResource PRIVATE
         CafeDefaultFont.cpp
         )
 
-target_include_directories(CemuResource PRIVATE ../)
+target_include_directories(CemuResource PUBLIC "../")
+
+target_link_libraries(CemuResource PRIVATE CemuComponents)

--- a/src/util/CMakeLists.txt
+++ b/src/util/CMakeLists.txt
@@ -1,7 +1,5 @@
 project(CemuUtil)
 
-include_directories(".")
-
 file(GLOB_RECURSE CPP_FILES *.cpp)
 file(GLOB_RECURSE H_FILES *.h)
 
@@ -11,5 +9,16 @@ set_property(TARGET CemuUtil PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CON
 
 target_precompile_headers(CemuUtil PRIVATE ../Common/precompiled.h)
 
-target_link_libraries(CemuUtil CemuCommon CemuConfig)
-target_include_directories(CemuUtil PRIVATE ../)
+target_include_directories(CemuUtil PUBLIC "../")
+
+target_link_libraries(CemuUtil PRIVATE
+	CemuCommon
+	CemuConfig
+	Boost::headers
+	Boost::nowide
+	OpenSSL::Crypto
+)
+
+if (ENABLE_WXWIDGETS)
+	target_link_libraries(CemuUtil PRIVATE wx::base wx::core)
+endif()

